src/ui/common/src/components/pages/function/id/index.tsx

```diff
@@ -0,0 +1,409 @@
+import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Link, List, ListItem } from '@mui/material';
+import Accordion from '@mui/material/Accordion';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import AccordionSummary from '@mui/material/AccordionSummary';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import { BlobReader, TextWriter, ZipReader } from '@zip.js/zip.js';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link as RouterLink, useNavigate, useParams } from 'react-router-dom';
+
+import DefaultLayout from '../../../../components/layouts/default';
+import LogViewer from '../../../../components/LogViewer';
+import MultiFileViewer from '../../../../components/MultiFileViewer';
+import { boolArtifactNodeIcon } from '../../../../components/workflows/nodes/BoolArtifactNode';
+import { dictArtifactNodeIcon } from '../../../../components/workflows/nodes/DictArtifactNode';
+import { imageArtifactNodeIcon } from '../../../../components/workflows/nodes/ImageArtifactNode';
+import { jsonArtifactNodeIcon } from '../../../../components/workflows/nodes/JsonArtifactNode';
+import { numericArtifactNodeIcon } from '../../../../components/workflows/nodes/NumericArtifactNode';
+import { stringArtifactNodeIcon } from '../../../../components/workflows/nodes/StringArtifactNode';
+import { tableArtifactNodeIcon } from '../../../../components/workflows/nodes/TableArtifactNode';
+import {
+  handleGetArtifactResults,
+  handleGetOperatorResults,
+  handleGetWorkflow,
+  selectResultIdx,
+} from '../../../../reducers/workflow';
+import { AppDispatch, RootState } from '../../../../stores/store';
+import { ArtifactType } from '../../../../utils/artifacts';
+import UserProfile from '../../../../utils/auth';
+import { getPathPrefix } from '../../../../utils/getPathPrefix';
+import { exportFunction } from '../../../../utils/operators';
+import { LoadingStatusEnum } from '../../../../utils/shared';
+import DetailsPageHeader from '../../components/DetailsPageHeader';
+import { LayoutProps } from '../../types';
+
+type FunctionDetailsPageProps = {
+  user: UserProfile;
+  Layout?: React.FC<LayoutProps>;
+  maxRenderSize?: number;
+};
+
+const listStyle = {
+  width: '100%',
+  maxWidth: 360,
+  bgcolor: 'background.paper',
+};
+
+// Checked with file size=313285391 and handles that smoothly once loaded. However, takes a while to load.
+const FunctionDetailsPage: React.FC<FunctionDetailsPageProps> = ({
+  user,
+  Layout = DefaultLayout,
+  maxRenderSize = 100000000,
+}) => {
+  const dispatch: AppDispatch = useDispatch();
+  const navigate = useNavigate();
+  const [inputsExpanded, setInputsExpanded] = useState<boolean>(true);
+  const [outputsExpanded, setOutputsExpanded] = useState<boolean>(true);
+  const [files, setFiles] = useState({
+    '': {
```

Comment:
Instead of just passing in the object here, you should probably define a type above and pass that into useState.

e.g.

```typescript
const FilePreview = {
  path: '',
  language: 'plaintext',
  content: ''
};

const [files, setFiles] = useState<FilePreview>(null);
```